src/persist/src/operators/replay.rs

```diff
@@ -37,15 +39,20 @@ where
     fn replay(
         &self,
         snapshot: Result<DecodedSnapshot<K, V>, Error>,
+        as_of_frontier: &Antichain<u64>,
```

Comment:
doesn't this mean the contract of the operator needs to be that it forwards all data to this frontier before emitting it? I don't see where that happens. we should also add a test for this

same q for the listen_source operator.

the persisted_source operator shouldn't need to do any forwarding itself, but probably wants a debug assert (and maybe also a test?)

Reply:
it's not 100% clear to me wether it needs to do that. We have code in the source rendering code that does that: https://github.com/MaterializeInc/materialize/blob/0d0ae6f876307dcfd168aef25478bf98b6d64ade/src/dataflow/src/render/sources.rs#L605. Maybe @frankmcsherry has an opinion on this: should the replay source advance times on the data it emits to the `as_of`, or should it just verify that its since/compaction frontier is not beyond the `as_of` and rely on the rest of the source pipeline to advance the timestamp?

I'll add the tests for sure, though!